File: sorting-snippets/insertionsort/insertionsort.py
collection = [1,4,8,21,6,1,12,5]

def insertion_sort(collection):
    for i in range(1, len(collection)):
        while i > 0 and collection[i-1] > collection[i]:
            collection[i], collection[i-1] = collection[i-1], collection[i]
            i -= 1
    return collection

print(insertion_sort(collection))

File: sorting-snippets/insertionsort/insertionsort.go
package main

import (
	"fmt"
)

func insertionSort(array []int) []int {
	for i:= 0; i < len(array); i++ {
		for i > 0 && array[i-1] > array[i] {
			array[i], array[i-1] = array[i-1], array[i]
			i--
		}
	}
	return array
}

func main() {
	a := []int{1,4,8,21,6,1,12,5}
	fmt.Println(insertionSort(a))
}
